File: python3/1855.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        i = len(nums1) - 1
        j = len(nums2) - 1
        md = 0
        while i >= 0:
            while i <= j and nums1[i] > nums2[j]:
                j -= 1
            if j - i > md:
                md = j - i
            i -= 1
        return md

File: c/1855.c
int maxDistance(int* nums1, int n1, int* nums2, int n2) {
    int i = n1 - 1, j = n2 - 1, md = 0;
    while (i >= 0) {
        while (i <= j && nums1[i] > nums2[j])
            j--;
        if (j - i > md)
            md = j - i;
        i--;
    }
    return md;
}
